src/pelicanfs/core.py

```diff
@@ -43,19 +44,28 @@ class PelicanFileSystem(AsyncFileSystem):
     
     def __init__ (
             self,
-            directorUrl,
+            federationDiscoveryUrl,
+            dirReads = False,
+            specifiedCache = None,
             asynchronous = False,
             loop = None
     ):
         
         # The internal filesystem
         self.httpFileSystem = fshttp.HTTPFileSystem(asynchronous=asynchronous, loop=loop)
 
-        # Ensure the director url ends with a "/"
+        # Perform federation discovery
+        metadata_json = self._discover_federation_metadata(federationDiscoveryUrl)
+
+        # Ensure the director url has a '/' at the end
+        directorUrl = metadata_json['director_endpoint']
```

Comment:
This will throw an exception if the `director_endpoint` key is missing.  Handle the case explicitly and create a specific exception for invalid discovery metadata.